data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/PitWorker.java

```diff
@@ -191,11 +195,14 @@ private void processIndex(final SourcePartition<OpenSearchIndexProgressState> op
 
             searchWithSearchAfterResults.getDocuments().stream().map(Record::new).forEach(record -> {
                 try {
+                    final long documentBytes = objectMapper.writeValueAsBytes(record.getData().getJsonNode()).length;
```

Comment:
Writing this to bytes is going to add a performance hit.

Also, there is a possibility that this will be somewhat different than the input since we are looking at the `Event` here rather than the actual JSON document.

Reply:
The JsonNode in the event is essentially SearchResults/hit/source JsonNode. I tried to normalize the document unit for bytesReceived and bytesProcessed so that user can make a comparison. I am open to other alternative unit.

Reply:
I see that we set it to the document `_source` here:

https://github.com/opensearch-project/data-prepper/blob/b80b565583c998e43a809a12e43dca9f9a20f6f0/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java#L296

That may or may not be the best metric, but I think it can work.

Reply:
I do not have a better solution for performance hit yet since the SDK client returns ObjectNode...